pkg/snowflake/gen_test.go

```diff
@@ -60,6 +66,24 @@ func BenchmarkEncode(b *testing.B) {
 	}
 }
 
+func BenchmarkNext(b *testing.B) {
+	g := New(10)
+
+	for i := 0; i < b.N; i++ {
+		g.Next()
+	}
+}
+
+func BenchmarkNextParallel(b *testing.B) {
+	g := New(1)
+
+	b.RunParallel(func(pb *testing.PB) {
+		for pb.Next() {
+			g.Next()
+		}
+	})
+}
```

Comment:
Looks like the call to g.Next is being elided out, you may want to make sure the compiler knows not to do that.

Reply:
It actually isn't in this case (confirmed with -gcflags=-S), but I'll update to make sure it doesn't happen in the future as best as I'm able.